File: 0LeetCodeProblems/[SolutionLeetcode]/LeetCode_Solutions-main/two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> pairs = new HashMap();
        
        for (int i = 0; i < nums.length; i++) {
            int diff = target - nums[i];

            if (pairs.get(diff) != null) {

                return new int[]{pairs.get(diff), i};
            }
            
            pairs.put(nums[i], i);
        }
        return new int[]{0,0};
    }
}

File: 1LeetCodeProblems/[SolutionLeetcode]/LeetCode_Solutions-main/two-sum/two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let seen = {}
    
    for (let i = 0; i < nums.length; i++) {
        let diff = target - nums[i]
        
        if (diff in seen) {
            return [seen[diff], i]
        }
        
        seen[nums[i]] = i
    }
};